From: Takashi Iwai <tiwai@suse.de>
Date: Thu, 3 Dec 2009 05:12:01 +0000
Subject: [PATCH 10/56] net: Fix Yukon-2 Optima TCP offload setup

commit b338682dc5c20e8ff986e58407bdb6e3a3e3f0a3 upstream.

Fix the TCP offload setup for Yukon-2 Optima.
It requires SKY2_HW_NE_LE flag unlike Ultra 2.

Signed-off-by: Takashi Iwai <tiwai@suse.de>
Signed-off-by: David S. Miller <davem@davemloft.net>
---
 drivers/net/sky2.c |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/drivers/net/sky2.c b/drivers/net/sky2.c
index 3943d89..050e6b5 100644
--- a/drivers/net/sky2.c
+++ b/drivers/net/sky2.c
@@ -2968,8 +2968,13 @@ static int __devinit sky2_init(struct sky2_hw *hw)
 		break;
 
 	case CHIP_ID_YUKON_UL_2:
+		hw->flags = SKY2_HW_GIGABIT
+			| SKY2_HW_ADV_POWER_CTL;
+		break;
+
 	case CHIP_ID_YUKON_OPT:
 		hw->flags = SKY2_HW_GIGABIT
+			| SKY2_HW_NEW_LE
 			| SKY2_HW_ADV_POWER_CTL;
 		break;
 
-- 
1.7.1

